pkg/builder/controller.go

```diff
@@ -255,10 +256,20 @@ func (blder *Builder) doWatch() error {
 	for _, w := range blder.watchesInput {
 		allPredicates := append([]predicate.Predicate(nil), blder.globalPredicates...)
 		allPredicates = append(allPredicates, w.predicates...)
-		if err := blder.ctrl.Watch(w.src, w.eventhandler, allPredicates...); err != nil {
-			return err
+		watchSource := w.src
+
+		// If the source of this watch is of type *source.Kind, project it.
+		if srckind, ok := watchSource.(*source.Kind); ok {
+			typeForSrc, err := blder.project(srckind.Type, w.objectProjection)
+			if err != nil {
+				return err
+			}
+			watchSource = &source.Kind{Type: typeForSrc}
```

Comment:
Please re-use the existing object and only overwrite the `Type`, right now we drop for example the cache and in theory it could get more attributes in the future

Reply:
Done